### PR TITLE
add app_launcher docs and basic options descriptions

### PR DESCRIPTION
**Pull request by JimmyCozza**
Refer to #130 

I was actually trying to implement something exactly like this myself without knowing it already existed.  Hopefully adding the docs will help someone else find it on the bling website.

Couple things in this documentation.

1) I wasn't sure the proper way to give a description of every option, some I left blank because I don't understand what the options do.  (save_history, the `commands` options, and the rubato option)
2) I noticed a typo on the `shrink_width` option, but I didn't want to fix it without running it by your team first.  The correct move would be to rename from `shirnk_width` to `shrink_width` but I don't know the downstream effects or how you normally disseminate breaking changes like that, so I left it as `shirnk_width` and documented it as such.  Should I add a more clear note in these docs letting end users know that it's not a typo in the docs, rather the real option name?
3) Do you want more images to demonstrate the options?  I wasn't sure so I just used the image @JavaCafe01 posted a little bit ago in one of the PRs (https://github.com/BlingCorp/bling/pull/103#issuecomment-960128253).

Let me know what I can improve on this, and thank you all so much for bling.

### CODE REVIEW

**Comment by Nooo37** [comment]
Very cool that you did that, thank you

> don't understand what the options do. (save_history, the commands options, and the rubato option)

Not a huge user of the module myself. My guesses are that Rubato is probably there to make the launcher "fly" into the screen and should be a table `{ x = rubato_animation_x, y = rubato_animation_y }`. `search_commands` should be a boolean that determines whether the app launcher should not only match the name of the app but also the commandline command with which it would start (https://lazka.github.io/pgi-docs/Gio-2.0/interfaces/AppInfo.html#Gio.AppInfo.get_commandline). So `skip_commands` would be the same as `skip_names` but for commandline options (eg maybe something like skip all that have a "--whatever" flag). Maybe @Kasper24 or @JavaCafe01 can provide more insights?

> rename from shirnk_width to shrink_width but I don't know the downstream effects or how you normally disseminate breaking changes

I would say, just rename the option. We don't do releases so there is no good way to fix something like that

> Do you want more images to demonstrate the options?

One image is enough, it's just there so people can get an idea. We handle it like that in other modules too :+1:

File: widget/app_launcher/init.lua [hunk 1]
```diff
@@ -770,14 +770,14 @@ local function new(args)
     args.default_app_icon_name = args.default_app_icon_name or nil
     args.default_app_icon_path = args.default_app_icon_path or nil
     args.icon_theme = args.icon_theme or nil
-    args.icons_size = args.icons_size or nil
+    args.icon_size = args.icon_size or nil
```
Comment (JimmyCozza) [inline widget/app_launcher/init.lua:773]: I noticed the arg was called `icons_size` while the code was looking for `icon_size` (on line 589)

File: docs/widgets/app_launcher.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    default_app_icon_name = "standard.svg"                            -- Sets default app icon name for apps without icon names
+    default_app_icon_path = "~/icons/"                                -- Sets default app icon path for apps without icon paths
+    icon_theme = "application"                                        -- Set icon theme
+    icon_size = 24                                                    -- Set icon size
+
+    type = "dock"                                                     -- awful.popup type ("dock", "desktop", "normal"...).  See awesomewm docs for more detail
+    show_on_focused_screen = true                                     -- Should app launcher show on currently focused screen
+    screen = awful.screen                                             -- Screen you want the launcher to launch to
+    placement = awful.placement.top_left                              -- Where launcher should be placed ("awful.placement.centered").
+    rubato = { x = rubato_animation_x, y = rubato_animation_y }       -- Rubato animation to apply to launcher
```
Comment (JimmyCozza) [inline docs/widgets/app_launcher.md:64]: I'll have to defer to you on this rubato option.  I could not get anything to work in my testing, but I have never used rubato before and I don't know when I'm going to have the time/energy to go down that rabbit hole.

**Comment by JimmyCozza** [comment]
@Kasper24 Question for you about how you intended for the skip_commands option to work.

Was it intended to allow me to skip some granular levels of commandline command like Noo037 mentioned above?

Let's say I want to have some Thunar commands show in the list, but not all of them.  I normally have 
- `thunar %F` (Thunar File Manager)
- `thunar --bulk-rename %F` (Bulk Rename)
- `thunar-settings` (File Manager Settings)
- `thunar-volman-settings` (Removable Drives and Media)
in my app launcher.

Say I don't want the Bulk Rename option to show, should I be able to pass in `--bulk-rename %F` to skip that app?
or
Should I pass in `thunar --bulk-rename %F`to skip it?

Right now, neither of those is what actually happens for me, so maybe I still don't understand the intent.

**Comment by Nooo37** [comment]
Kasper seems to be awol for the time being. Thank you for your contribution of making his work usable for other people too, very much appreciated

**Pull request merged**